tests/test_normailzation.py

```diff
@@ -0,0 +1,9 @@
+from sms_toolkit.messages.normalization import normalize_optimistic_gsm_characters
+
+
+class TestMessageNormalization:
+    def test_normalize_unicode(self, sample_unicode_string):
+        assert normalize_optimistic_gsm_characters(sample_unicode_string) == u"This is my MANAGEMENT VOICE."
+
+    def test_normalize_byte_string(self, sample_byte_string):
+        assert normalize_optimistic_gsm_characters(sample_byte_string) == u"This is my MANAGEMENT VOICE."
```

Comment:
Do you think its worth adding another test method that passes in a subset character map and confirming that only the characters in the map are replaced?

Reply:
Also maybe one that includes non-replaceable non-gsm characters, specifically an emoji?

Reply:
I think I follow your second comment but I'm confused by the first. Would you providing a small example of the type of string you would expect in that case?

Reply:
I remember now. The subset would be just to test that kwarg you added to pass a specific character mapping, rather than just using the default. No worries if you wanted to skip the test, just wanted to close the thread on it.

Reply:
Actually I think I'm just gonna remove that kwarg. I was trying to make the function pure but I just realized it's still not pure anyway.